src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -0,0 +1,1475 @@
+    placement.setPosition(BoxPosition);
+    pcBox->Placement.setValue(placement);
+
+    auto CutObject = doc->getObject(CutYName);
+    // there should be a cut, but maybe the user deleted it meanwhile
+    if (!CutObject) {
+        Base::Console().Warning((std::string("SectionCut warning: there is no ")
+            + std::string(CutYName) + std::string(", trying to recreate it\n")).c_str());
+        // recreate the box
+        startCutting();
+        return;
+    }
+
+    // if there is another cut, we must recalculate it too
+    // we only need to check for Z since the hierarchy is always Z->Y->X
+    if (hasBoxZ) {
+        auto CutFeatureZ = doc->getObject(CutZName);
+        Part::Cut* pcCutZ = static_cast<Part::Cut*>(CutFeatureZ);
+        pcCutZ->recomputeFeature(true);
+    }
+    else {
+        Part::Cut* pcCut = static_cast<Part::Cut*>(CutObject);
+        pcCut->recomputeFeature(true);
+    }
+}
+
+void SectionCut::onFlipZclicked()
+{
+    // there might be no document
+    if (!Gui::Application::Instance->activeDocument()) {
+        noDocumentActions();
+        return;
+    }
+    // refresh objects and return in case the document was changed
+    if (doc != Gui::Application::Instance->activeDocument()->getDocument()) {
+        onRefreshCutPBclicked();
+        return;
+    }
+    // we must move the box in z-direction by its Height
+        // get the cut box
+    auto CutBox = doc->getObject(BoxZName);
+    // there should be a box, but maybe the user deleted it meanwhile
+    if (!CutBox) {
+        Base::Console().Warning((std::string("SectionCut warning: there is no ")
+            + std::string(BoxZName) + std::string(", trying to recreate it\n")).c_str());
+        // recreate the box
+        startCutting();
+        return;
+    }
+    Part::Box* pcBox = static_cast<Part::Box*>(CutBox);
+    // get its placement and size
+    Base::Placement placement = pcBox->Placement.getValue();
+    Base::Vector3d BoxPosition = placement.getPosition();
+    // flip the box
+    if (ui->flipZ->isChecked())
+        BoxPosition.z = BoxPosition.z + pcBox->Height.getValue();
+    else
+        BoxPosition.z = BoxPosition.z - pcBox->Height.getValue();
+    placement.setPosition(BoxPosition);
+    pcBox->Placement.setValue(placement);
+
+    auto CutObject = doc->getObject(CutZName);
+    // there should be a cut, but maybe the user deleted it meanwhile
+    if (!CutObject) {
+        Base::Console().Warning((std::string("SectionCut warning: there is no ")
+            + std::string(CutZName) + std::string(", trying to recreate it\n")).c_str());
+        // recreate the box
+        startCutting();
+        return;
+    }
+    Part::Cut* pcCut = static_cast<Part::Cut*>(CutObject);
+    pcCut->recomputeFeature(true);
+}
+
+// refreshes the list of document objects and the visible objects
+void SectionCut::onRefreshCutPBclicked()
+{
+    // get document
+    doc = Gui::Application::Instance->activeDocument()->getDocument();

```

Comment:
@donovaly This check is unsafe. Because if there is not active document then `activeDocument()` already returns null and thus will cause a segmentation fault.

Reply:
done in #6360